core/sugarTraversals.ml

```diff
@@ -559,9 +559,14 @@ class map =
           let _x_i1 = o#phrase _x_i1 in Table ((_x, _x_i1))
 
     method funlit : funlit -> funlit =
-      fun (_x, _x_i1) ->
-        let _x = o#list (fun o -> o#list (fun o -> o#pattern)) _x in
-        let _x_i1 = o#phrase _x_i1 in (_x, _x_i1)
+      fun f ->
+        match f with
+          | NormalFunlit (_x, _x_i1) ->
+            let _x = o#list (fun o -> o#list (fun o -> o#pattern)) _x in
+            let _x_i1 = o#phrase _x_i1 in NormalFunlit (_x, _x_i1)
+          | SwitchFunlit (pat, body) ->
+            let pat = o#list (fun o -> o#list (fun o -> o#pattern)) pat in
+            let body = o#list (fun o (p, c) -> let p = o#pattern p in let c = o#phrase c in (p, c)) body in SwitchFunlit (pat, body)
```

Comment:
This is currently a bit hard to read without the proper line breaks:

```suggestion
          | SwitchFunlit (pat, body) ->
            let pat = o#list (fun o -> o#list (fun o -> o#pattern)) pat in
            let body =
              o#list (fun o (p, c) ->
                let p = o#pattern p in
                let c = o#phrase c in (p, c)) body in
            SwitchFunlit (pat, body)
```

Reply:
Done. However, there are comments at the beginning of `sugarTraversals` saying that it should be automatically generated by Camlp4 (but I had to handly update it). Is readability a problem here?

Reply:
I think those comments are (a) very old and (b) no longer accurate - in particular camlp4 is no longer used at all by Links.  We discussed trying to reinstate the auto-generation of visitors/traversals a while ago (#255 ) but decided not to; we might revisit that now that Links has mostly been rewritten to use standard datatypes instead of polymorphic variants.

It would probably be reasonable to update the comment to state that this file is no longer autogenerated.